### PR TITLE
Allow 4.x callable to continue working in 5.x.

### PR DESCRIPTION
**Pull request by dereuromark**
When coming from a 4.x app with events and event system, the classes that implement __invoke are not working anymore.

Given that on()/off() accept callable, the Closure should be replaced with callable for normalizeHandlers() or at least callable be added, too.

I marked it as regression coming from 4.x as this is also not documented in the migration guide and makes apps trip over this when upgrading.

### CODE REVIEW

File: src/Event/EventManager.php [hunk 1]
```diff
@@ -226,17 +226,19 @@ protected function _detachSubscriber(EventListenerInterface $subscriber, ?string
...
      * @param \Cake\Event\EventListenerInterface $subscriber Event subscriber
-     * @param \Closure|array|string $handlers Event handlers
+     * @param \Closure|callable|array|string $handlers Event handlers
      * @return array
      */
-    protected function normalizeHandlers(EventListenerInterface $subscriber, Closure|array|string $handlers): array
-    {
+    protected function normalizeHandlers(
+        EventListenerInterface $subscriber,
+        Closure|callable|array|string $handlers
```
Comment (ADmad) [inline src/Event/EventManager.php]: The `Closure` is redundant, every closure is a callable.

**Comment by ADmad** [comment]
Would be nice to have a test for this too to avoid regression.

File: tests/test_app/TestApp/TestCase/Event/InvokableEventTestListener.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace TestApp\TestCase\Event;
+
+use Cake\Event\EventInterface;
+use Cake\Event\EventListenerInterface;
+
+/**
+ * Mock class used to test event dispatching
+ */
+class InvokableEventTestListener implements EventListenerInterface
```
Comment (LordSimal) [inline tests/test_app/TestApp/TestCase/Event/InvokableEventTestListener.php]: why not use an anonymous class?

Comment (dereuromark) [reply tests/test_app/TestApp/TestCase/Event/InvokableEventTestListener.php]: We can. I just am oldschool I guess - with like real test files ;)

**Review by LordSimal — APPROVED** [review]

**Review by ADmad — APPROVED** [review]

**Comment by markstory** [comment]
Nice. I agree we should relax types where it makes upgrading difficult and restriction doesn't  improve the developer experience of using the framework.

**Pull request merged**